commit 3639d08e2f36f76e9a626c60b534c7fe204f329c
Author: Peter Eisentraut <peter@eisentraut.org>
Date:   Thu Jun 20 11:36:38 2024 +0200

    pg_dump: Fix weird error message composition
    
    The previous way could make it look like "stdin" was the actual input
    file name.  Write it as two separate messages instead.

diff --git a/src/bin/pg_dump/filter.c b/src/bin/pg_dump/filter.c
index 3fb93c5f155..5815cd23748 100644
--- a/src/bin/pg_dump/filter.c
+++ b/src/bin/pg_dump/filter.c
@@ -159,14 +159,16 @@ pg_log_filter_error(FilterStateData *fstate, const char *fmt,...)
 
 	va_start(argp, fmt);
 	vsnprintf(buf, sizeof(buf), fmt, argp);
 	va_end(argp);
 
-	pg_log_error("invalid format in filter read from \"%s\" on line %d: %s",
-				 (fstate->fp == stdin ? "stdin" : fstate->filename),
-				 fstate->lineno,
-				 buf);
+	if (fstate->fp == stdin)
+		pg_log_error("invalid format in filter read from standard input on line %d: %s",
+					 fstate->lineno, buf);
+	else
+		pg_log_error("invalid format in filter read from file \"%s\" on line %d: %s",
+					 fstate->filename, fstate->lineno, buf);
 }
 
 /*
  * filter_get_keyword - read the next filter keyword from buffer
  *
